app/models/page.rb

```diff
@@ -94,4 +96,18 @@ def guidance_markdown_length_and_tags
       errors.add(:guidance_markdown, :unsupported_markdown_syntax, message: "can only contain formatting for links, subheadings(##), bulleted listed (*), or numbered lists(1.)")
     end
   end
+
+  def destroy_secondary_skip_conditions
+    return if goto_conditions.empty?
+
+    # We want to delete the secondary skip for the page at the start of the route
+    # That association isn't in the database, so we need to dig it out
+    # TODO: what if the page owning the routes has more than two routes?
+    goto_conditions
+      .filter { |condition| condition.check_page_id == condition.routing_page_id }
+      .map(&:check_page)
+      .flat_map(&:check_conditions)
+      .filter { |condition| condition.check_page_id != condition.routing_page_id }
+      .each(&:destroy!)
```

Comment:
Nice! These are hairy but hopefully we can come up with something better in the future which captures the what we are querying for.

Reply:
Yeah I fear I've created some tech debt here 😅